Doctrine/Phpcr/BaseSuggestionProvider.php

```diff
@@ -56,4 +57,26 @@ public function getManagerForClass($class)
     {
         return $this->managerRegistry->getManagerForClass($class);
     }
+
+    /**
+     * Finds the parent route document by concatenating the basepaths with the 
+     * requested path.
+     *
+     * @return null|\Traversable
+     */
+    protected function findParentRoute($requestedPath)
+    {
+        $manager = $this->getManagerForClass('Symfony\Cmf\Bundle\RoutingBundle\Doctrine\Phpcr\Route');
+
+        foreach ($this->routeBasePaths as $basepath) {
+            $parentPath = PathHelper::getParentPath($basepath.$requestedPath);
+
+            $parentRoute = $manager->find(null, $parentPath);
```

Comment:
from a performance point of view, i suspect that it would be better to first build the list of parent path candidates and then try to find all of them in one single call. i think the overhead of preparing all parent paths is small compared to the overhead of several calls into the phpcr. the chances that we find more than one result is pretty small.